src/components/UNSTABLE__Pagination/__tests__/UNSTABLE_Pagination.spec.js

```diff
@@ -0,0 +1,413 @@
+
+    // Expect page sizer select to not be in the document:
+    expect(
+      document.getElementById(
+        'security--unstable-pagination__page-sizer__input-1'
+      )
+    ).not.toBeInTheDocument();
+  });
+
+  test('should display `pageRangeText` only (without a page select) if page selector child not provided', () => {
+    const { getByText } = render(
+      <UNSTABLE__Pagination
+        pageSize={5}
+        pageSizes={[5]}
+        totalItems={10}
+        pageRangeText={(current, total) => `test ${current} of ${total} pages`}
+      />
+    );
+
+    // Expect `pageRangeText` as-is, with 1 being the current page
+    // and 2 being the total pages, based on the a page size of 5 for 10 total items.
+    expect(getByText(/test 1 of 2 pages/i)).toBeInTheDocument();
+  });
+
+  test('should not display total pages or total items if `pagesUnknown` is `true`, even if `totalItems` is provided', () => {
+    const { getByText, queryByText, rerender } = render(
+      <UNSTABLE__Pagination
+        pageSize={5}
+        pageSizes={[5]}
+        // Note that `pagesUnknown` has been set to `true`:
+        pagesUnknown
+        // Note that `totalItems` is not provided and is therefore `undefined`.
+      >
+        {({ currentPage, onSetPage, totalPages }) => (
+          <PageSelector
+            currentPage={currentPage}
+            labelText="test page selector"
+            onChange={event => onSetPage(event.target.value)}
+            totalPages={totalPages}
+          />
+        )}
+      </UNSTABLE__Pagination>
+    );
+
+    // Expect default "of 2 pages" text not to be shown,
+    // because `pagesUnknown={true}`:
+    expect(queryByText(/of 2 pages/i)).not.toBeInTheDocument();
+
+    // Expect default text "1-5 items" with 5 being derived from page size.
+    // The max in the range is NOT derived from the total items:
+    expect(getByText(/1–5 items/i)).toBeInTheDocument();
+
+    rerender(
+      <UNSTABLE__Pagination
+        pageSize={5}
+        pageSizes={[5]}
+        // Note that `pagesUnknown` has been set to `true`:
+        pagesUnknown
+        // Note that `totalItems` is now provided:
+        totalItems={10}
+      >
+        {({ currentPage, onSetPage, totalPages }) => (
+          <PageSelector
+            currentPage={currentPage}
+            labelText="test page selector"
+            onChange={event => onSetPage(event.target.value)}
+            totalPages={totalPages}
+          />
+        )}
+      </UNSTABLE__Pagination>
+    );
+
+    // Still expect default "of 2 pages" text not to be shown,
+    // because `pagesUnknown={true}` (despite the fact that `totalItems` is provided):
+    expect(queryByText(/of 2 pages/i)).not.toBeInTheDocument();
+
+    // Still expect default text "1-5 items" with 5 being derived from page size,
+    // not the total items (despite the fact that `totalItems` is provided):
+    expect(getByText(/1–5 items/i)).toBeInTheDocument();
+  });
+});
```

Comment:
@diego-codes if you have a moment I was wondering if you could take a look at this last test and let me know what you think...

I'm trying to check if that certain text is display if `pagesUnknown` is `true`, *whether or not* `totalItems` is provided. I think it's important to check both situations, where `totalItems` is provided or not, because `totalItems` is this case is superceded by `pagesUnknown`.

I went with a `rerender` here, in the same test. At first I just rerendered with a valid `totalItems` value, thinking that my previous props would be carried over. But then when things didn't look right, I opted to redefined *everything* in this rerender. Does that make sense to do or would you recommend I split the test?

Reply:
Hi @jendowns! These two scenarios are two use cases so I would write two separate tests. It's better to think of tests as use cases instead of "features" because we end up with bloated tests like so.

Reply:
Thank you that makes sense! Honestly at this point I'm not sure when/why I'd ever use `rerender`? 😅

Reply:
Most test cases don't require it. When you truly need it, you'll know 😆… It's useful when the change of a prop over time is required to demonstrate a test case.